pkg/storj/streamid_test.go

```diff
@@ -6,28 +6,28 @@ package storj_test
 import (
 	"testing"
 
-	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	"storj.io/storj/internal/testrand"
 	"storj.io/storj/pkg/storj"
 )
 
 func TestStreamID_Encode(t *testing.T) {
-	_, err := storj.StreamIDFromString("likn43kilfzd")
-	assert.Error(t, err)
-
-	_, err = storj.StreamIDFromBytes([]byte{1, 2, 3, 4, 5})
-	assert.Error(t, err)
-
 	for i := 0; i < 10; i++ {
-		streamID := testrand.StreamID()
+		expectedSize := testrand.Intn(255)
+		streamID := testrand.StreamID(expectedSize)
 
 		fromString, err := storj.StreamIDFromString(streamID.String())
-		assert.NoError(t, err)
+		require.NoError(t, err)
+		require.Equal(t, streamID.String(), fromString.String())
```

Comment:
Aren't we overusing `require` over `assert`?

`require.NoError` is usually good as it prevents the test from panicking. But `require.Equal` interrupts the tests too early. If we have a failing test (especially flaky one) the result from the rest of the checks might be very valuable in analyzing the issue.

Reply:
personally, I prefer to fail-fast but I agree that with more complex test it might be useful to have results from multiple assertions. In this case rerunning test is a matter of seconds so having only first error is good enough for me.